### PR TITLE
Draft: bug fixing

### PR DESCRIPTION
**Pull request by yanyag**
Support writing and parsing "filesAnalyzed" into XML/JSON/YAML. 
Fix parsers to support multiple packages per document.
All the tests are passing.

### CODE REVIEW

File: spdx/parsers/jsonyamlxml.py [hunk 1]
```diff
@@ -1181,11 +1185,31 @@ def parse_pkg_down_location(self, pkg_down_location):
...
 
+    def parse_pkg_files_analyzed(self, pkg_files_analyzed):
+        """
+        Parse Package files analyzed
+        - pkg_files_analyzed: Python boolean or str/unicode
+        """
+        # Files Analyzed optional
+        if pkg_files_analyzed is None:
+            return
+        if isinstance(pkg_files_analyzed, six.string_types) or isinstance(pkg_files_analyzed, bool):
```
Comment (tardyp) [inline spdx/parsers/jsonyamlxml.py:1196]: according to the spec, https://spdx.github.io/spdx-spec/3-package-information/
filesAnalysed can only be a boolean.

So I would have stored the conversion to bool.

Also, we are removing six in #179 , so please don't use six for new code/

File: spdx/parsers/jsonyamlxml.py [hunk 2]
```diff
@@ -1181,11 +1185,31 @@ def parse_pkg_down_location(self, pkg_down_location):
...
+                self.more_than_one_error("PKG_FILES_ANALYZED")
+        else:
+            self.value_error("PKG_FILES_ANALYZED", pkg_files_analyzed)
+
     def parse_pkg_verif_code_field(self, pkg_verif_code_field):
         """
         Parse Package verification code dict
         - pkg_verif_code_field: Python dict('value':str/unicode, 'excludedFilesNames':list)
         """
+        if self.document.packages[-1].files_analyzed == False:
```
Comment (tardyp) [inline spdx/parsers/jsonyamlxml.py:1211]:
```suggestion
        if not self.document.packages[-1].files_analyzed:
```

Comment (tardyp) [reply spdx/parsers/jsonyamlxml.py:1211]: why testing on packages[-1]? Sounds like a hack assuming the current package being parser is the last one.

I think we may rather want to do this test in the caller function

File: spdx/parsers/tagvalue.py [hunk 3]
```diff
@@ -1183,6 +1183,7 @@ def p_package_name(self, p):
                 value = p[2].decode(encoding="utf-8")
             else:
                 value = p[2]
+            self.builder.reset_package()
```
Comment (tardyp) [inline spdx/parsers/tagvalue.py:1186]: not sure why this change. Can you please explain?

File: spdx/writers/jsonyamlxml.py [hunk 4]
```diff
@@ -116,16 +116,20 @@ def create_package_info(self, package):
         package_object["SPDXID"] = self.spdx_id(package.spdx_id)
         package_object["name"] = package.name
         package_object["downloadLocation"] = package.download_location.__str__()
-        package_object["packageVerificationCode"] = self.package_verification_code(
-            package
-        )
+        if package.files_analyzed in [True, None]:
```
Comment (tardyp) [inline spdx/writers/jsonyamlxml.py:119]: as the default value for this field is True, we should initialize the field to True.

this test is really not obvious, and will lead to bugs in the future

File: spdx/writers/jsonyamlxml.py [hunk 5]
```diff
@@ -116,16 +116,20 @@ def create_package_info(self, package):
...
+                package
+            )
         package_object["licenseConcluded"] = self.license(package.conc_lics)
         package_object["licenseInfoFromFiles"] = list(
             map(self.license, package.licenses_from_files)
         )
         package_object["licenseDeclared"] = self.license(package.license_declared)
         package_object["copyrightText"] = package.cr_text.__str__()
 
+        if package.has_optional_field("files_analyzed"):
```
Comment (tardyp) [inline spdx/writers/jsonyamlxml.py:130]: as per spec: 3.8.3 Cardinality: Optional, one. If omitted, the default value of true is assumed.

so imho we should really initialize this to True, an only write if it is set to False

```suggestion
        if package.files_analyzed:
```

**Review by tardyp — CHANGES_REQUESTED** [review]

**Comment by tardyp** [comment]
hi @yanyag can you please rebase this PR?

now that the six removal PR is merged, there are a few conflicts.

**Comment by tardyp** [comment]
hi @yanyag  @itraviv ,

Please remember you still have this PR open that needs rebasing. It still contains merge conflicts.

Now the PR contains more fixes than initially pushed, and it becoming less trivial.
I think it will need more unit tests, with 2.2 file examples which proves that your code is actually working.

**Comment by pombredanne** [comment]
Gentle ping :)

**Comment by itraviv** [comment]
hi, we will probably need some time until we'll be able to have this MR ready due to capacity. will ping when ready.

**Comment by pombredanne** [comment]
gentle ping :) Any update?

**Comment by armintaenzertng** [comment]
As this PR is still in draft mode and has not been receiving vital updates for over a year, I suggest closing it.

@yanyag, any objections?

**Comment by nicoweidner** [comment]
Closing this due to inactivity, please ping if it should be reopened.

**Pull request closed without merging**